### ISSUE TITLE
# of requests per connection

### ISSUE THREAD

**Issue by mathiasbl** [MicrosoftDocs/powerapps-docs#1019]
I've read this page a dozen times by now but only this week I noticed that a single user can have 52 concurrent connections and each connection can do 6000 requests according to the documentation. Or at least what I understand of it.

This would mean a single user can open up 52 connections and get a throughput of 52 x 6000 requests. In reality this is not the case.

Measure | Description | Limit per web server
-- | -- | --
Number of requests | The actual number of requests made **per connection**. | 6000
Execution time | The combined duration for all connections using the same user account | 20 minutes (1200 seconds)
Number of connections | **The number of connections using the same user account** | 52

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 3fd657fb-b2cb-a85a-7243-41d7cc30dc87
* Version Independent ID: 4ed6833a-668b-ff05-97da-286fb6d68394
* Content: [Service Protection API Limits (Common Data Service) - PowerApps](https://docs.microsoft.com/en-us/powerapps/developer/common-data-service/api-limits)
* Content Source: [powerapps-docs/developer/common-data-service/api-limits.md](https://github.com/MicrosoftDocs/powerapps-docs/blob/live/powerapps-docs/developer/common-data-service/api-limits.md)
* Service: **powerapps**
* GitHub Login: @JimDaly
* Microsoft Alias: **jdaly**

**Comment by JimDaly** [comment]
API requests are limited in three ways:
 - Number of requests
 - Execution time
 - Number of connections

Any one of them will cause a 429 error to be thrown.
If you attempt 52 concurrent connections, you shouldn't expect that each of them will be able to perform 6000 requests without hitting the Execution Time limit.

You may find that having more connections will give you higher throughput, but only if the requests are small and don't require too much execution time.

As called out in the topic, the specific limits are provided to inform about how the limits are applied per web server, but the key take-away from this topic is to make sure your code is resilient and can let the server tell you how fast you can send requests based on the Retry-After value.

**Comment by mathiasbl** [comment]
Hi @JimDaly, thanks for the quick response

I actually created a client tool to test these burst limits and I also expected to hit the execution time limit but I always got the 6000 request per 5min limit. This makes me believe the statement 'The actual number of requests made per connection =  6000' is false. The 'per connection' does not apply. 

These were my test results:
I was able to create 20.000 to 22.0000 accounts in a window of 5 minutes but every single request after this the Retry-After value had values from 5 to 8 minutes so I guess this would average to ~6000 to 8000 requests in a window of 5 minutes.

What I'm really after is some guidance how to set up an integration with a custom websites where the load varies and could hit the 6000 requests per 5 minute. You don't want your custom website to be effected by a dependency api. Any guidance on this? Should we use multiple application users circumvent the api limits?
